models/object_detection_yolox/demo.py

```diff
@@ -0,0 +1,132 @@
+    parser.add_argument('--model', type=str, default='yolox_s.onnx', help="Path to the model")
+    parser.add_argument('--input_type', type=str, default='image', help="Input types: image or video")
+    parser.add_argument('--image_path', type=str, default='test2.jpg', help="Image path")
+    parser.add_argument('--video_path', type=str, default='sample.mp4', help="Video path")
+    parser.add_argument('--confidence', default=0.5, type=float, help='Class confidence')
+    parser.add_argument('--nms', default=0.5, type=float, help='Enter nms IOU threshold')
+    parser.add_argument('--obj', default=0.5, type=float, help='Enter object threshold')
+    parser.add_argument('--save', '-s', type=str, default=False, help='Set true to save results. This option is invalid when using camera.')
+    args = parser.parse_args()
+
+    model_net = YoloX(modelPath= args.model, confThreshold=args.confidence, nmsThreshold=args.nms, objThreshold=args.obj)
+    
+    if (args.input_type=="image"):
+        srcimg = cv2.imread(args.image_path)
+
+        image = srcimg
+        a = time.time()
+        preds = model_net.infer(srcimg)
+        b = time.time()
+        print('Inference_Time:'+str(b-a)+' secs')
+
+        srcimg = vis(preds, image)
+
+        cv2.namedWindow(args.image_path, cv2.WINDOW_AUTOSIZE)
+        cv2.imshow(args.image_path, srcimg)
+        cv2.waitKey(0)
+
+        if args.save:
+            print('Resutls saved to result.jpg\n')
+            cv2.imwrite('result.jpg', image)
+
+    else:
+        print("Press 1 to stop video capture")
+        cap = cv2.VideoCapture(0)
+        tm = cv2.TickMeter()
+        total_frames = 0
+        frame_list = []
+        Video_save = False
+        if(args.save):
+            Video_save = True
+
+        while cv2.waitKey(1) < 0:
+            hasFrame, frame = cap.read()
+            if not hasFrame:
+                print('No frames grabbed!')
+                break
+
+            frame = cv2.flip(frame, 1)
+            srcimg = frame
+
+            # Inference
+            tm.start()
+            preds = model_net.infer(srcimg)
+            tm.stop()
+
+            srcimg = vis(preds, srcimg)
+
+            total_frames += 1
+            fps=tm.getFPS()
+
+            if fps > 0:
+                fps_label = "FPS: %.2f" % fps
+                cv2.putText(srcimg, fps_label, (10, 25), cv2.FONT_HERSHEY_SIMPLEX, 1, (0, 0, 255), 2)
+
+            cv2.imshow("output", srcimg)
+
+            if cv2.waitKey(1) > -1:
+                print("Stream terminated")
+                break
+
+            if(args.save):
+                srcimg = cv2.cvtColor(srcimg, cv2.COLOR_BGR2RGB)
+                frame_list.append(srcimg)
+
+        if(Video_save):
+            clip = moviepy.video.io.ImageSequenceClip.ImageSequenceClip(frame_list, fps=fps)
+            clip.write_videofile('Webcam_result.mp4')
```

Comment:
OpenCV provides `VideoWriter` to save videos. Here is an example: https://stackoverflow.com/a/29317298/6769366

Reply:
replaced it with VideoWriter